src/account/UpgradeableModularAccount.sol

```diff
@@ -109,21 +109,21 @@ contract UpgradeableModularAccount is
 
     /// @notice Initializes the account with a set of modules
     /// @param modules The modules to install
-    /// @param manifestHashes The manifest hashes of the modules to install
+    /// @param manifests The manifests of the modules to install
     /// @param moduleInstallDatas The module install datas of the modules to install
     function initialize(
```

Comment:
One part I really liked about the previous PluginManifest design is that doing plugin.pluginManifest() is the cheapest possible option (compared to calldata, or storage) to pass data into the installation flow. Curious to hear your thoughts on whether its worth including an option to "install plugin with it's recommended config, as-is" that would pull the manifest from the plugin. 

In general, I'm not a fan of one-offs, but this should be a huge calldata savings in the average case and would require alterations to the spec (changing the `initialize` call format) and think it might be worth including in the spec and in RI

Reply:
Do you think it's possible to make it an optional part of the spec for modules to implement? "Modules MAY implement a `pluginManifest()` function..." (maybe we should rename it to `recommendedManifest` to signal that it isn't actually a guarantee to be respected by accounts, and is optional)

This way, common plugins can still provide the configs needed to install efficiently.

And then the rest can be achieved (h/t @adamegyed came up with this) via a plugin with permission to call install functions.

Down the line we could explore splitting and simplifying installation though!

Reply:
I agree there should be some mechanism like `recommendedManifest` or `defaultManifest`, as there will inevitably be install configurations that result in a malfunctioning module. Auditors could then review modules based on configurations defined in the given configurations

Reply:
Yep, I still think the plugins should report manifests as their intended configuration, if they define functions that use manifests for install (like execution functions, hooks associated with an exec selector, or an interface ID representing multiple execution functions).

> doing plugin.pluginManifest() is the cheapest possible option

I don't think this is the case – performing the call, the ABI encode/decode, and the hash verification, is necessarily more execution gas than just performing the ABI decode from calldata. The only case where the on-chain call could be cheaper is if the calldata cost is very high, at which point implementing the manifest reading mechanism on-chain makes sense, because both storage and execution gas would be relatively cheaper than the calldata cost.